stream-chat-android-client/src/main/java/io/getstream/chat/android/client/utils/ThreadLocalDelegate.kt

```diff
@@ -0,0 +1,19 @@
+package io.getstream.chat.android.client.utils
+
+import kotlin.properties.ReadOnlyProperty
+import kotlin.reflect.KProperty
+
+internal class ThreadLocalDelegate<T : Any>(
```

Comment:
Cool. DefaultDateTypeAdapter from Gson uses synchronization for thread safety, but this approach definitely better.

Reply:
I never thought to check it out, but that's actually a great idea. Looking at it now, they synchronize on the `DateFormat` instance(s), which is better than the method level synchronization that we had.

Still, thread local makes this even cleaner, so I think it's worth it.

Reply:
Yep, I like this solution